src/renderer/components/NewServerModal.tsx

```diff
@@ -428,6 +480,105 @@ class NewServerModal extends React.PureComponent<Props, State> {
                         {this.getServerNameMessage()}
                         {this.getServerURLMessage()}
                     </div>
+                    {this.props.editMode &&
```

Comment:
This file handles the whole modal, and the different toggles look pretty similar to each other. I wonder if we should divide it a bit more into different components.

Reply:
They're very similar components which is why they're the same. This file predates both of us by a fair bit.
As above, would love to give the Desktop App front-end some love :)